test/integration/agent/networkpolicy_metrics_test.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agent
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/watch"
+	k8stesting "k8s.io/client-go/testing"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/component-base/metrics/testutil"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/controller/networkpolicy"
+	"github.com/vmware-tanzu/antrea/pkg/agent/metrics"
+	"github.com/vmware-tanzu/antrea/pkg/apis/networking/v1beta1"
+	"github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned"
+	"github.com/vmware-tanzu/antrea/pkg/client/clientset/versioned/fake"
+)
+
+const testNamespace = "ns1"
+
+type antreaClientGetter struct {
+	clientset versioned.Interface
+}
+
+func (g *antreaClientGetter) GetAntreaClient() (versioned.Interface, error) {
+	return g.clientset, nil
+}
+
+func newTestController() (*networkpolicy.Controller, *fake.Clientset) {
+	clientset := &fake.Clientset{}
+	ch := make(chan v1beta1.PodReference, 100)
+	controller := networkpolicy.NewNetworkPolicyController(&antreaClientGetter{clientset}, nil, nil, "node1", ch)
+	return controller, clientset
+}
+
+func newNetworkPolicy(uid string, from, to, appliedTo []string, services []v1beta1.Service) *v1beta1.NetworkPolicy {
+	networkPolicyRule1 := v1beta1.NetworkPolicyRule{
+		Direction: v1beta1.DirectionIn,
+		From:      v1beta1.NetworkPolicyPeer{AddressGroups: from},
+		To:        v1beta1.NetworkPolicyPeer{AddressGroups: to},
+		Services:  services,
+	}
+	return &v1beta1.NetworkPolicy{
+		ObjectMeta:      v1.ObjectMeta{UID: types.UID(uid), Name: uid, Namespace: testNamespace},
+		Rules:           []v1beta1.NetworkPolicyRule{networkPolicyRule1},
+		AppliedToGroups: appliedTo,
+	}
+}
+
+func getNetworkPolicyWithMultipleRules(uid string, from, to, appliedTo []string, services []v1beta1.Service) *v1beta1.NetworkPolicy {
```

Comment:
Any reason why the name is different newNetworkPolicy as in why not newNetworkPolicyWithMultipleRules?

Reply:
Sorry, there is no reason. I'll make them consistent.